.github/workflows/ci.yml

```diff
@@ -11,78 +11,54 @@ env:
 
 jobs:
   build_and_test:
-    name: Deluxe
+    strategy:
+      matrix:
+        features: ["--all-features", "--no-default-features"]
     runs-on: ubuntu-latest
     timeout-minutes: 45
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          override: true
-      - run: cargo build -r --all-features --workspace
-      - run: cargo test -r --all-features --workspace
-
-  build_and_test_no_default:
-    name: Deluxe No Default Features
-    runs-on: ubuntu-latest
-    timeout-minutes: 45
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          override: true
-      - run: cargo build -r --no-default-features --workspace
-      - run: cargo test -r --no-default-features --workspace
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
+      - run: cargo build -r ${{ matrix.features }} --workspace
+      - run: cargo test -r ${{ matrix.features }} --workspace
 
   rustfmt:
     name: Deluxe Rustfmt
     runs-on: ubuntu-latest
     timeout-minutes: 45
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          profile: minimal
-          toolchain: stable
-          override: true
           components: rustfmt
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo fmt --all --check
 
   clippy:
     name: Deluxe Clippy
     runs-on: ubuntu-latest
     timeout-minutes: 45
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
```

Comment:
there's also an argument for pinning the version of clippy to a recent nightly.
using a pinned version prevents new errors popping up in PRs due to new lints that aren't related to the changes in the PR.
the downside is you have to remember to manually bump it occasionally

Reply:
i've gone ahead and pinned it.